NonLocFEM/thermal_problems_2d.hpp

```diff
@@ -33,11 +33,16 @@ heat_equation_solution_2d<T> solve_thermal_2d_problem(
     std::shared_ptr<mesh::mesh_2d<T>>& mesh,
     const parameters_2d<T>& parameters,
     const thermal::thermal_boundaries_conditions_2d<T>& boundaries_conditions,
-    const config::thermal_auxiliary_data<T>& auxiliary) {
-    auto solution = nonlocal::thermal::stationary_heat_equation_solver_2d<I>(
-        mesh, parameters, boundaries_conditions, 
-        [value = auxiliary.right_part](const std::array<T, 2>& x) constexpr noexcept { return value; },
-        auxiliary.energy
+    const config::thermal_auxiliary_data_2d<T>& auxiliary) {
+    const stationary_equation_parameters_2d<T> auxiliary_data {
+        .right_part = [value = std::get<1>(auxiliary.right_part)](const std::array<T, 2>& x) constexpr noexcept { return value(x); },
```

Comment:
Что такое std::get<1> максимально не понятно, лучше явно указать тип, который собираешься извлекать
```suggestion
        .right_part = [value = std::get<spatial_dependency<T, 2>>(auxiliary.right_part)](const std::array<T, 2>& x) constexpr noexcept { return value(x); },
```